File: app/src/main/java/com/arcgistest/activities/base/BaseActivity.java
package com.arcgistest.activities.base;


import android.support.v7.app.AppCompatActivity;

import com.arcgistest.app.ARCGIApplication;
import com.arcgistest.di.components.ApplicationComponent;
import com.arcgistest.di.modules.ActivityModule;

public abstract class BaseActivity extends AppCompatActivity {

    protected ApplicationComponent getAppComponent() {
        return ((ARCGIApplication) getApplication()).getApplicationComponent();
    }

    protected ActivityModule getActivityModule() {
        return new ActivityModule(this);
    }
}

File: app/src/main/java/com/arcgistest/activities/base/BaseActivity.kt
package com.arcgistest.activities.base

import android.support.v7.app.AppCompatActivity
import com.arcgistest.app.ARCGIApplication
import com.arcgistest.di.components.ApplicationComponent
import com.arcgistest.di.modules.ActivityModule

abstract class BaseActivity : AppCompatActivity() {

    protected val appComponent: ApplicationComponent
        get() = (application as ARCGIApplication).applicationComponent

    protected val activityModule: ActivityModule
        get() = ActivityModule(this)
}